internal/service/projectapikey/resource_project_api_key.go

```diff
@@ -69,64 +68,47 @@ func Resource() *schema.Resource {
 	}
 }
 
-type APIProjectAssignmentKeyInput struct {
-	ProjectID string   `json:"desc,omitempty"`
-	RoleNames []string `json:"roles,omitempty"`
-}
-
-const errorNoProjectAssignmentDefined = "could not obtain a project id as no assignments are defined"
-
 func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	if err := validateUniqueProjectIDs(d); err != nil {
```

Comment:
if we are adding validation, have you considered doing this in the read / plan phase?

Reply:
i did that first, but it was a problem because existing resources would fail in Read so nothing could be done with those resources if terraform state is not edited